### PR TITLE
Update changelog

### PR DESCRIPTION
**Pull request by rahulporuri**
Add changelog entries for the 4.7.0 release, which were missing, and
for changes introduced after the 4.7.0 release.

### CODE REVIEW

**Comment by dpinte** [comment]
Errors are related to the ipython integration.

**Comment by rahulporuri** [comment]
@dpinte merged master into this PR and updated the changelog with an entry for the recent PR. take another look?

File: CHANGES.txt [hunk 1]
```diff
@@ -2,6 +2,42 @@
  Envisage CHANGELOG
 ====================
 
+Dev
```
Comment (mdickinson) [inline CHANGES.txt]: I'd suggest just removing this heading.

**Comment by rahulporuri** [comment]
@mdickinson CI is green now. looks like it was an intermittent failure.

**Pull request merged**